### ISSUE TITLE
Unavailable Package Source Slows Install

### ISSUE THREAD

**Issue by Feasoron** [NuGet/Home#1682]
If there is a package source in the nuget config that is unavailable for any reason, installing or updating a package will hang for a long time (~10 minutes is the longest I saw) at the "Gathering Dependencies" stage. The package does not need to have anything to do with the unavailable source for this to occur.

**Comment by yishaigalatzer** [comment]
NuGet is trying to get packages from all sources, it cannot know upfront if the source includes a new version of that packages or any of its dependencies. 

Just to verify, this is installing with known version or install the latest version?

Can you give us a more specific example?

**Comment by yishaigalatzer** [comment]
We have an idea to detect slow sources and pop a UI that would suggest disabling them.

**Comment by Feasoron** [comment]
This was happening for me when installing the latest version or updating to the latest. Any call of install-package would do this. I also saw it when updating AutoMapper to the latest. A popup to disable the source would be great - my concern isn't that NuGet is checking the package sources as much as it is that I received no indication as to what was slowing the install. It was only through trial and error that I found the source that was an issue.
